internal/handlers/common/params.go

```diff
@@ -391,6 +374,26 @@ func multiplyNumbers(v1, v2 any) (any, error) {
 	}
 }
 
+// multiplyLongSafely returns the multiplication of two int64 values.
+// It handles int64 overflows, and returns errLongExceeded error on one.
+//
+// Please prefer to use multiplyNumbers instead of this function.
+func multiplyLongSafely(v1, v2 int64) (int64, error) {
+	switch {
+	case v1 == 0 || v2 == 0 || v1 == 1 || v2 == 1:
```

Comment:
I see why `v2 == 0` is here - to avoid division by zero later, but why do the other values are here too? Are those cases also special?

Reply:
We check for 1 and 0 for every value, because it's possible to multiply those with MinInt64. Without this check, function returns overflow on operations like `MinInt64 * 1` or `0 * MinInt64`, etc.

Reply:
Should I leave a comment explaining that? It was confusing either to me, and I've written this code 2 hours ago...

Reply:
although now I'm thinking about `-1` values either. It's not ready for review yet sorry

Reply:
@noisersup I remember you said you want to try `-1` and some other cases, is it still actual? I see the tests that handle `-1`, but according to the logic in this function, `errLongExceeded` will be returned for them, right?

Reply:
Not exactly, in the first `multiplyLongSafely` case we check only for `0` and `1`, as those are only values possible to multiply correctly with `MinInt64`.

All other values compared to any other value are handled after the switch case, and all overflows should be detected there.